src/buildx/install.ts

```diff
@@ -230,4 +237,59 @@ export class Install {
     }
     return releases[version];
   }
+
+  public async installReleaseFromFork(url: string): Promise<string> {
+    const targetFile: string = os.platform() == 'win32' ? 'docker-buildx.exe' : 'docker-buildx';
+    let [base, version] = url.split('#');
+    if (!version) {
+      return '';
+    }
+    if (!version.startsWith('v')) {
+      core.info(`Only supports v-prefixed releases`);
+      return '';
+    }
+    version = version.substring(1);
+
+    if (!base.startsWith('https://github.com/')) {
+      core.info(`Only supports github releases. Actual: ${base}`);
+      return '';
+    }
+
+    let [owner, repo] = base.substring('https://github.com/'.length).split('/');
+    if (!owner || !repo) {
+      core.info(`Only supports github releases. Actual: ${base}, owner: ${owner}, repo: ${repo}`);
+      return '';
+    }
+    if (repo.endsWith('.git')) {
+      repo = repo.substring(0, repo.length - 4);
+    }
+
+    core.info(`Checking for releases in ${owner}/${repo}`);
+    const octokit = this._octokit;
+    const release = await octokit.request('GET https://api.github.com/repos/{owner}/{repo}/releases/tags/{tag}', {
```

Comment:
`api.github.com` might not be right for orgs in GHES. Better to use https://github.com/docker/actions-toolkit/blob/ca519e1aa876dc868f4d313c6c9cf25b3edc0be1/src/github.ts#L48-L50

Reply:
Also let's avoid static API refs and instead use octokit REST methods to fetch the release asset. Smth like https://github.com/crazy-max/ghaction-virustotal/blob/f7edea8b65ab5e752aa4aaf1f1fc487efcc5d282/src/github.ts#L48-L83

Reply:
hmmmm... i don't think that's quite right. currently setup-buildx-action asks people to specify their version as `https://github.com/owner/repo#tag`.

```
    - name: Set up Docker Buildx
      uses: docker/setup-buildx-action@v2
      with:
        use: true
        version: https://github.com/fork-owner/buildx#v0.12.0
```

How would this work if we wanted to use the structured api? are you imagining that we'd take an object as input?

```
    - name: Set up Docker Buildx
      uses: docker/setup-buildx-action@v2
      with:
        use: true
        release:
          owner: "fork-owner"
          repo: "buildx"
```

?

similarly, `process.env.GITHUB_API_URL` won't work -- that points to the repo the user is running the action against, not the repo they want to download buildx from. right?

Reply:
I don't think we need extra inputs, just:

```yaml
    - name: Set up Docker Buildx
      uses: docker/setup-buildx-action@v2
      with:
        use: true
        version: https://github.com/crazy-max/buildx/releases/tag/v0.10.4%2Blocalstate2
```

If version is a valid smever or a valid GH release url then `download`, otherwise `build`.

Reply:
> similarly, `process.env.GITHUB_API_URL` won't work -- that points to the repo the user is running the action against, not the repo they want to download buildx from. right?

I don't think the release target matters but the fqdn is as it needs to access the right endpoint to query the API in orgs using GHES and not `api.github.com` as it would fail. `repos/{owner}/{repo}/releases/tags/{tag}` target will still succeed to find the release on GHES afaik.